rts/Lua/LuaSyncedRead.cpp

```diff
@@ -3652,14 +3630,75 @@ int LuaSyncedRead::GetProjectilesInRectangle(lua_State* L)
 			if (!LuaUtils::IsProjectileVisible(L, pro))
 				continue;
 
-			lua_pushnumber(L, pro->id);
-			lua_rawseti(L, -2, arrayIndex++);
+			lua_pushnumber(L, static_cast<lua_Number>(pro->id));
```

Comment:
I dislike these casts. Perhaps we should update the build process to enable some sort of "ignore warnings" flag for these?